tob-web/src/app/general-data.service.ts

```diff
@@ -342,53 +354,50 @@ export class GeneralDataService {
   }
 
   translateClaimDescription(credTypeId, claimName, defVal?) {
-    let credLang = this.getCredentialTypeLanguageKey(credTypeId, 'claim_descriptions');
-    return credLang.then(values => {
-      let lang = this.language;
-      let ret = undefined;
-      if(lang in values) {
-        ret = values[lang][claimName];
-      }
-      if(ret === undefined)
-        ret = defVal;
-      return ret;
-    });
+    let credLang = this.getCredentialTypeLanguageKey(credTypeId, "claim_descriptions");
+    return credLang.pipe(
+      map(values => {
+        let lang = this.language;
+        let ret = undefined;
+        if (values && claimName in values) {
+          ret = values[claimName][lang];
+        }
+        if (ret === undefined) ret = defVal;
+        return ret;
+      })
+    );
   }
 
   translateClaimLabel(credTypeId, claimName, defVal?) {
-    let credLang = this.getCredentialTypeLanguageKey(credTypeId, 'claim_labels');
-    return credLang.then(values => {
-      let lang = this.language;
-      let ret = undefined;
-      if(lang in values) {
-        ret = values[lang][claimName];
-      }
-      if(ret === undefined)
-        ret = defVal;
-      return ret;
-    });
+    let credLang = this.getCredentialTypeLanguageKey(credTypeId, "claim_labels");
+    return credLang.pipe(
+      map(values => {
+        let lang = this.language;
+        let ret = undefined;
+        if (values && claimName in values) {
+          ret = values[claimName][lang];
+        }
+        if (ret === undefined) ret = defVal;
+        return ret;
+      })
+    );
   }
 
   translateCategoryLabel(credTypeId, catType, catValue) {
-    let credLang = this.getCredentialTypeLanguageKey(credTypeId, 'category_labels');
+    let credLang = this.getCredentialTypeLanguageKey(credTypeId, "category_labels");
     let lbl = `category.${catType}.${catValue}`;
     return credLang.pipe(
       map(values => {
         let lang = this.language;
         let ret = undefined;
-        if(values && lang in values) {
-          let labels = values[lang];
-          if(labels && catType in labels) {
-            ret = labels[catType][catValue];
-          }
+        if (values && catType in values) {
```

Comment:
This was changes to be consistent with the code in the above functions. The API call is the same, so the returned data structure must be the same. I was not able to create `category_labels` to test this with though.